src/t8_schemes/t8_default/t8_dpyramid_bits.c

```diff
@@ -0,0 +1,1498 @@
+        t->type = p->type == 3 ? 1 : 0;
+      }
+      else {
+        t->x = p->x * T8_DTRI_ROOT_BY_DPYRAMID_ROOT;
+        t->type = p->type == 3 ? 1 : 0;
+      }
+    }
+    T8_ASSERT (t->type == 0 || t->type == 1);
+  }
+}
+
+int
+t8_dpyramid_extrude_face (const t8_element_t * face, t8_dpyramid_t * p,
+                          const int root_face)
+{
+  T8_ASSERT (0 <= root_face && root_face < T8_DPYRAMID_FACES);
+
+  int                 extruded_face;
+  if (root_face == 4) {
+    /* Pyramids on the bottom are always type 6 pyramids at the bottom. We need to
+     * scale the coordinates, since a quad and a pyra can have different root-len,
+     * depending on their maxlvl.*/
+    p4est_quadrant_t   *q = (p4est_quadrant_t *) face;
+    /*Typecast to int64, we multiply two (possible at max) int32*/
+    p->x = ((int64_t) q->x * T8_DPYRAMID_ROOT_LEN) / P4EST_ROOT_LEN;
+    p->y = ((int64_t) q->y * T8_DPYRAMID_ROOT_LEN) / P4EST_ROOT_LEN;
+    p->z = 0;
+    p->type = T8_DPYRAMID_ROOT_TPYE;
+    p->level = q->level;
+    return root_face;
+  }
+  else {
+    /*t->y gives the height of the pyramid, t->x gives the p->x or p->y, depending on
+     * the root_face. The other coordinate is determined by the root_face.*/
+    t8_dtri_t          *t = (t8_dtri_t *) face;
+    p->z = ((int64_t) t->y * T8_DPYRAMID_ROOT_LEN) / T8_DTRI_ROOT_LEN;
+    /* level is the same */
+    p->level = t->level;
+    switch (root_face) {
+    case 0:
+      p->x = p->z;
+      /*Typecast to int64, we multiply two (possible at max) int32*/
+      p->y = ( (int64_t)t->x * T8_DPYRAMID_ROOT_LEN) / T8_DTRI_ROOT_LEN;
+      break;
+    case 1:
+      p->x = T8_DPYRAMID_ROOT_LEN - T8_DPYRAMID_LEN (p->level);
+      p->y = ((int64_t)t->x * T8_DPYRAMID_ROOT_LEN) / T8_DTRI_ROOT_LEN;
+      break;
+    case 2:
+      p->x = ( (int64_t)t->x * T8_DPYRAMID_ROOT_LEN) / T8_DTRI_ROOT_LEN;
+      p->y = p->z;
+      break;
+    case 3:
+      p->x = ( (int64_t)t->x * T8_DPYRAMID_ROOT_LEN) / T8_DTRI_ROOT_LEN;
+      p->y = T8_DPYRAMID_ROOT_LEN - T8_DPYRAMID_LEN (p->level);
+      break;
+    default:
+      SC_ABORT_NOT_REACHED ();
+    }
+    /*Description of triangles extruding to a pyramid */
+    if ((t->y == (t->x & t->y)) && t->type == 0) {
+      /*type zero in a pyramid extend to a pyramid */
+      p->type = 6;
+      extruded_face = root_face;
+    }
+    else {
+      /*type 0 not in a pyramid extend to a tetrahedron */
+      p->type = t8_dpyramid_tritype_rootface_to_tettype[t->type][root_face];
+      extruded_face = t8_dpyramid_tritype_rootface_to_face[t->type][root_face];
+    }
+  }
+  T8_ASSERT (0 <= p->level && p->level <= T8_DPYRAMID_MAXLEVEL);
+  /*return the face-number of the extruded face */
+  return extruded_face;
+}
+
+int
+t8_dpyramid_child_id_unknown_parent (const t8_dpyramid_t * p,
+                                     t8_dpyramid_t * parent)
+{
+  T8_ASSERT (p->level >= 0);
+  if (p->level == 0)
+    return 0;
```

Comment:
Would it not be more sensible to return -1 in this case? Thus, an invalid child id?
Since if p is level 0 it is not the child of anyone.
This may also be important to the tell the calling function that parent was not filled with values.

Reply:
Done